### PR TITLE
ci: config test workflow and setup vitest

### LINKED ISSUE #87: Add vitest config
## Description

We need automated tests as soon as possible in the core. Tests ensure not only quality but also accelerate development, a strong culture of automated testing makes the team more agile and with more bug-free deliveries.

We already have some tests ( e.g. https://github.com/expressots/expressots/blob/main/packages/core/test/application.spec.ts), but they are not running correctly.

The goal of this issue is to fix the config and enable tests to run, it also should use Vitest instead of Jest, they are almost the same in terms of syntax. However, Vitest is much faster.

Ref.
https://vitest.dev/
https://vitest.dev/config/
https://dev.to/mbarzeev/from-jest-to-vitest-migration-and-benchmark-23pl

## Requirements

- [x] vite and vitest must be listed in the project references;
- [x] all tests must run;
- [x] tests must run in the CI.

## Definition of Done

- [ ] assert that tests are running correctly using vitest;
- [ ] assert that the action is correctly running in the PR.

### PR DESCRIPTION
**Pull request by Amorim33**
closes #87 

# Pull Request Guidelines

Our guidelines for submitting a pull request.

## Before submitting a Pull Request, please make sure you have verified the following:

- [x] The commit message follows our guidelines:
  - A good commit message should be two things: meaningful and concise. It should not contain every single detail, describing each changed line—we can see all the changes in Git—but, at the same time, it should say enough to avoid ambiguity.
  - We use Microverse's commit message convention
  - The convention stablish that a commit message has to be in the present tense, imperative and lowercase.
  - Example: `fix typo in README.md`
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

- Tests using `jest`;
- Tests not running.

Issue Number: #87 

## What is the new behavior?

- Tests using `vitest`;
- Tests running locally and in the CI.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications below.

### CODE REVIEW

File: packages/core/test/application.spec.ts
Comment (rsaz) [inline packages/core/test/application.spec.ts:1]: I think you need the import to make vi.spyOn to work

```import { vi } from "vitest";```

File: packages/core/test/console.spec.ts
Comment (rsaz) [inline packages/core/test/console.spec.ts:1]: I think you need the import to make vi.spyOn to work

```import { vi } from "vitest";```

File: packages/core/test/env-validator.spec.ts
Comment (rsaz) [inline packages/core/test/env-validator.spec.ts:1]: I think you need the import to make vi.spyOn to work

```import { vi } from "vitest";```

**Review by rsaz — CHANGES_REQUESTED** [review]
Besides the import you need to review two points:
1 - eslint configuration there is an exclusion pattern to jest there thats needs to be readjusted to vitest now
2 - update the documentation in the section of Test: https://doc.expresso-ts.com/docs/overview/dependencies

image

Overall I think this initial setup is great. Once we get this right you can mode to the tasks below in another ticket:
1 - setup the code coverage as part of the pipeline based on the ci-cd repo I sent you
2 - implement testing snapshot strategy for all core library

Other than that, so far good job.

File: package.json [hunk 1]
```diff
@@ -86,13 +87,15 @@
     "@types/node": "^20.4.9",
```
Comment (rsaz) [inline package.json:87]: Need to remove `@types/jest`

File: packages/core/test/application.spec.ts
Comment (Amorim33) [reply packages/core/test/application.spec.ts:1]: Actualy, I think there is no need.
It should be set globally by vitest, just like in Jest.
However, in opoosition to jest, it is not set by default, we need to define `globals: true` in `vitest.config.ts`.

Ref.

https://vitest.dev/config/#globals

**Pull request merged**